### ISSUE TITLE
0.8.3: looks like `notfound` is not ready for sphinx >= 6.1.3

### ISSUE THREAD

**Issue by kloczek** [readthedocs/sphinx-notfound-page#217]
Look lik after upgrade to latest sphinx 6.1.4 pytest started failing
<details>

```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-sphinx-notfound-page-0.8.3-7.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-sphinx-notfound-page-0.8.3-7.fc35.x86_64/usr/lib/python3.8/site-packages
+ /usr/bin/pytest -ra -m 'not network'
============================= test session starts ==============================
platform linux -- Python 3.8.16, pytest-7.2.2, pluggy-1.0.0
rootdir: /home/tkloczko/rpmbuild/BUILD/sphinx-notfound-page-0.8.3, configfile: pytest.ini
collected 30 items

tests/test_urls.py ..F..FF.FF.........F.......x..                        [100%]

=================================== FAILURES ===================================
____________________________ test_default_settings _____________________________

app = <SphinxTestApp buildername='html'>
status = <_io.StringIO object at 0x7fea10d79ee0>
warning = <_io.StringIO object at 0x7fea10d79e50>

    @pytest.mark.sphinx(srcdir=srcdir)
    def test_default_settings(app, status, warning):
        app.build()
        path = app.outdir / '404.html'
        assert path.exists()
        content = open(path).read()

        chunks = [
            '<h1>Page not found</h1>',
            "Unfortunately we couldn't find the content you were looking for.",
            '<title>Page not found &#8212; Python  documentation</title>',

            # favicon and logo
            '<link rel="shortcut icon" href="/en/latest/_static/favicon.png"/>',
            '<img class="logo" src="/en/latest/_static/logo.svg" alt="Logo"/>',

            # sidebar URLs
            '<h1 class="logo"><a href="/en/latest/index.html">Python</a></h1>',
            '<form class="search" action="/en/latest/search.html" method="get">',
            '<li><a href="/en/latest/index.html">Documentation overview</a><ul>',

            # resources
            _get_css_html_link_tag('en', 'latest', 'alabaster.css'),
            _get_css_html_link_tag('en', 'latest', 'pygments.css'),
            '<link rel="stylesheet" href="/en/latest/_static/custom.css" type="text/css" />',
        ]

        for chunk in chunks:
>           assert chunk in content
E           assert '<link rel="shortcut icon" href="/en/latest/_static/favicon.png"/>' in '\n<!DOCTYPE html>\n\n<html lang="en">\n  <head>\n    <meta charset="utf-8" />\n    <meta name="viewport" content="wid..."https://github.com/bitprophet/alabaster">Alabaster 0.7.13</a>\n      \n    </div>\n\n    \n\n    \n  </body>\n</html>'

tests/test_urls.py:81: AssertionError
--------------------------- Captured stdout teardown ---------------------------
# testroot: root
# builder: html
# srcdir: /home/tkloczko/rpmbuild/BUILD/sphinx-notfound-page-0.8.3/tests/examples/default
# outdir: /home/tkloczko/rpmbuild/BUILD/sphinx-notfound-page-0.8.3/tests/examples/default/_build/html
# status:
Running Sphinx v6.1.3
building [mo]: targets for 0 po files that are out of date
writing output...
building [html]: targets for 3 source files that are out of date
updating environment: [new config] 3 added, 0 changed, 0 removed
reading sources... [100%] index
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] index
generating indices... genindex done
writing additional pages... search done
copying static files... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded.

The HTML pages are in tests/examples/default/_build/html.

# warning:

________________________ test_default_language_setting _________________________

app = <SphinxTestApp buildername='html'>
status = <_io.StringIO object at 0x7fea10b25a60>
warning = <_io.StringIO object at 0x7fea10b25f70>

    @pytest.mark.sphinx(
        srcdir=srcdir,
        confoverrides={
            'notfound_default_language': 'ja',
        },
    )
    def test_default_language_setting(app, status, warning):
        app.build()
        path = app.outdir / '404.html'
        assert path.exists()

        content = open(path).read()

        chunks = [
            # sidebar URLs
            '<h1 class="logo"><a href="/ja/latest/index.html">Python</a></h1>',
            '<form class="search" action="/ja/latest/search.html" method="get">',
            '<li><a href="/ja/latest/index.html">Documentation overview</a><ul>',

            # favicon and logo
            '<link rel="shortcut icon" href="/ja/latest/_static/favicon.png"/>',
            '<img class="logo" src="/ja/latest/_static/logo.svg" alt="Logo"/>',

            # resources
            _get_css_html_link_tag('ja', 'latest', 'alabaster.css'),
            _get_css_html_link_tag('ja', 'latest', 'pygments.css'),
            '<link rel="stylesheet" href="/ja/latest/_static/custom.css" type="text/css" />',
        ]

        for chunk in chunks:
>           assert chunk in content
E           assert '<link rel="shortcut icon" href="/ja/latest/_static/favicon.png"/>' in '\n<!DOCTYPE html>\n\n<html lang="en">\n  <head>\n    <meta charset="utf-8" />\n    <meta name="viewport" content="wid..."https://github.com/bitprophet/alabaster">Alabaster 0.7.13</a>\n      \n    </div>\n\n    \n\n    \n  </body>\n</html>'

tests/test_urls.py:148: AssertionError
--------------------------- Captured stdout teardown ---------------------------
# testroot: root
# builder: html
# srcdir: /home/tkloczko/rpmbuild/BUILD/sphinx-notfound-page-0.8.3/tests/examples/default
# outdir: /home/tkloczko/rpmbuild/BUILD/sphinx-notfound-page-0.8.3/tests/examples/default/_build/html
# status:
Running Sphinx v6.1.3
building [mo]: targets for 0 po files that are out of date
writing output...
building [html]: targets for 3 source files that are out of date
updating environment: [new config] 3 added, 0 changed, 0 removed
reading sources... [100%] index
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] index
generating indices... genindex done
writing additional pages... search done
copying static files... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded.

The HTML pages are in tests/examples/default/_build/html.

# warning:

_________________________ test_default_version_setting _________________________

app = <SphinxTestApp buildername='html'>
status = <_io.StringIO object at 0x7fea10c44430>
warning = <_io.StringIO object at 0x7fea10c44280>

    @pytest.mark.sphinx(
        srcdir=srcdir,
        confoverrides={
            'notfound_default_version': 'customversion',
        },
    )
    def test_default_version_setting(app, status, warning):
        app.build()
        path = app.outdir / '404.html'
        assert path.exists()

        content = open(path).read()

        chunks = [
            # sidebar URLs
            '<h1 class="logo"><a href="/en/customversion/index.html">Python</a></h1>',
            '<form class="search" action="/en/customversion/search.html" method="get">',
            '<li><a href="/en/customversion/index.html">Documentation overview</a><ul>',

            # favicon and logo
            '<link rel="shortcut icon" href="/en/customversion/_static/favicon.png"/>',
            '<img class="logo" src="/en/customversion/_static/logo.svg" alt="Logo"/>',

            # resources
            _get_css_html_link_tag('en', 'customversion', 'alabaster.css'),
            _get_css_html_link_tag('en', 'customversion', 'pygments.css'),
            '<link rel="stylesheet" href="/en/customversion/_static/custom.css" type="text/css" />',
        ]

        for chunk in chunks:
>           assert chunk in content
E           assert '<link rel="shortcut icon" href="/en/customversion/_static/favicon.png"/>' in '\n<!DOCTYPE html>\n\n<html lang="en">\n  <head>\n    <meta charset="utf-8" />\n    <meta name="viewport" content="wid..."https://github.com/bitprophet/alabaster">Alabaster 0.7.13</a>\n      \n    </div>\n\n    \n\n    \n  </body>\n</html>'

tests/test_urls.py:181: AssertionError
--------------------------- Captured stdout teardown ---------------------------
# testroot: root
# builder: html
# srcdir: /home/tkloczko/rpmbuild/BUILD/sphinx-notfound-page-0.8.3/tests/examples/default
# outdir: /home/tkloczko/rpmbuild/BUILD/sphinx-notfound-page-0.8.3/tests/examples/default/_build/html
# status:
Running Sphinx v6.1.3
building [mo]: targets for 0 po files that are out of date
writing output...
building [html]: targets for 3 source files that are out of date
updating environment: [new config] 3 added, 0 changed, 0 removed
reading sources... [100%] index
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] index
generating indices... genindex done
writing additional pages... search done
copying static files... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded.

The HTML pages are in tests/examples/default/_build/html.

# warning:

___________________________ test_urls_prefix_setting ___________________________

app = <SphinxTestApp buildername='html'>
status = <_io.StringIO object at 0x7fea109a4dc0>
warning = <_io.StringIO object at 0x7fea109a45e0>

    @pytest.mark.sphinx(
        srcdir=srcdir,
        confoverrides={
            'notfound_urls_prefix': '/language/version/',
        },
    )
    def test_urls_prefix_setting(app, status, warning):
        app.build()
        path = app.outdir / '404.html'
        assert path.exists()

        content = open(path).read()

        chunks = [
            # sidebar URLs
            '<h1 class="logo"><a href="/language/version/index.html">Python</a></h1>',
            '<form class="search" action="/language/version/search.html" method="get">',
            '<li><a href="/language/version/index.html">Documentation overview</a><ul>',

            # favicon and logo
            '<link rel="shortcut icon" href="/language/version/_static/favicon.png"/>',
            '<img class="logo" src="/language/version/_static/logo.svg" alt="Logo"/>',

            # resources
            _get_css_html_link_tag('language', 'version', 'alabaster.css'),
            _get_css_html_link_tag('language', 'version', 'pygments.css'),
            '<link rel="stylesheet" href="/language/version/_static/custom.css" type="text/css" />',
        ]

        for chunk in chunks:
>           assert chunk in content
E           assert '<link rel="shortcut icon" href="/language/version/_static/favicon.png"/>' in '\n<!DOCTYPE html>\n\n<html lang="en">\n  <head>\n    <meta charset="utf-8" />\n    <meta name="viewport" content="wid..."https://github.com/bitprophet/alabaster">Alabaster 0.7.13</a>\n      \n    </div>\n\n    \n\n    \n  </body>\n</html>'

tests/test_urls.py:243: AssertionError
--------------------------- Captured stdout teardown ---------------------------
# testroot: root
# builder: html
# srcdir: /home/tkloczko/rpmbuild/BUILD/sphinx-notfound-page-0.8.3/tests/examples/default
# outdir: /home/tkloczko/rpmbuild/BUILD/sphinx-notfound-page-0.8.3/tests/examples/default/_build/html
# status:
Running Sphinx v6.1.3
building [mo]: targets for 0 po files that are out of date
writing output...
building [html]: targets for 3 source files that are out of date
updating environment: [new config] 3 added, 0 changed, 0 removed
reading sources... [100%] index
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] index
generating indices... genindex done
writing additional pages... search done
copying static files... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded.

The HTML pages are in tests/examples/default/_build/html.

# warning:

________________________ test_urls_prefix_setting_none _________________________

app = <SphinxTestApp buildername='html'>
status = <_io.StringIO object at 0x7fea10910820>
warning = <_io.StringIO object at 0x7fea10910e50>

    @pytest.mark.sphinx(
        srcdir=srcdir,
        confoverrides={
            'notfound_urls_prefix': None,
        },
    )
    def test_urls_prefix_setting_none(app, status, warning):
        app.build()
        path = app.outdir / '404.html'
        assert path.exists()

        content = open(path).read()

        chunks = [
            # sidebar URLs
            '<h1 class="logo"><a href="/index.html">Python</a></h1>',
            '<form class="search" action="/search.html" method="get">',
            '<li><a href="/index.html">Documentation overview</a><ul>',

            # favicon and logo
            '<link rel="shortcut icon" href="/_static/favicon.png"/>',
            '<img class="logo" src="/_static/logo.svg" alt="Logo"/>',

            # resources
            _get_css_html_link_tag('', '', 'alabaster.css'),
            _get_css_html_link_tag('', '', 'pygments.css'),
            '<link rel="stylesheet" href="/_static/custom.css" type="text/css" />',
        ]

        for chunk in chunks:
>           assert chunk in content
E           assert '<link rel="shortcut icon" href="/_static/favicon.png"/>' in '\n<!DOCTYPE html>\n\n<html lang="en">\n  <head>\n    <meta charset="utf-8" />\n    <meta name="viewport" content="wid..."https://github.com/bitprophet/alabaster">Alabaster 0.7.13</a>\n      \n    </div>\n\n    \n\n    \n  </body>\n</html>'

tests/test_urls.py:276: AssertionError
--------------------------- Captured stdout teardown ---------------------------
# testroot: root
# builder: html
# srcdir: /home/tkloczko/rpmbuild/BUILD/sphinx-notfound-page-0.8.3/tests/examples/default
# outdir: /home/tkloczko/rpmbuild/BUILD/sphinx-notfound-page-0.8.3/tests/examples/default/_build/html
# status:
Running Sphinx v6.1.3
building [mo]: targets for 0 po files that are out of date
writing output...
building [html]: targets for 3 source files that are out of date
updating environment: [new config] 3 added, 0 changed, 0 removed
reading sources... [100%] index
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] index
generating indices... genindex done
writing additional pages... search done
copying static files... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded, 1 warning.

The HTML pages are in tests/examples/default/_build/html.

# warning:
WARNING: The config value `notfound_urls_prefix' has type `NoneType', defaults to `str'.

__________________________ test_sphinx_resource_urls ___________________________

app = <SphinxTestApp buildername='html'>
status = <_io.StringIO object at 0x7fea10ad9c10>
warning = <_io.StringIO object at 0x7fea10ad9af0>

    @pytest.mark.sphinx(srcdir=srcdir)
    def test_sphinx_resource_urls(app, status, warning):
        app.build()
        path = app.outdir / '404.html'
        assert path.exists()

        content = open(path).read()

        chunks = [
            # Sphinx's resources URLs
            _get_js_html_link_tag('en', 'latest', 'jquery.js'),
            _get_js_html_link_tag('en', 'latest', 'underscore.js'),
            _get_js_html_link_tag('en', 'latest', 'doctools.js'),
        ]

        # This file was added to all the HTML pages in Sphinx>=1.8. However, it was
        # only required for search page. Sphinx>=3.4 fixes this and only adds it on
        # search. See (https://github.com/sphinx-doc/sphinx/blob/v3.4.0/CHANGES#L87)
        if (1, 8) <= sphinx.version_info < (3, 4, 0):
            chunks.append(
                _get_js_html_link_tag('en', 'latest', 'language_data.js'),
            )

        for chunk in chunks:
>           assert chunk in content
E           assert '<script src="/en/latest/_static/jquery.js"></script>' in '\n<!DOCTYPE html>\n\n<html lang="en">\n  <head>\n    <meta charset="utf-8" />\n    <meta name="viewport" content="wid..."https://github.com/bitprophet/alabaster">Alabaster 0.7.13</a>\n      \n    </div>\n\n    \n\n    \n  </body>\n</html>'

tests/test_urls.py:565: AssertionError
--------------------------- Captured stdout teardown ---------------------------
# testroot: root
# builder: html
# srcdir: /home/tkloczko/rpmbuild/BUILD/sphinx-notfound-page-0.8.3/tests/examples/default
# outdir: /home/tkloczko/rpmbuild/BUILD/sphinx-notfound-page-0.8.3/tests/examples/default/_build/html
# status:
Running Sphinx v6.1.3
building [mo]: targets for 0 po files that are out of date
writing output...
building [html]: targets for 3 source files that are out of date
updating environment: [new config] 3 added, 0 changed, 0 removed
reading sources... [100%] index
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] index
generating indices... genindex done
writing additional pages... search done
copying static files... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded.

The HTML pages are in tests/examples/default/_build/html.

# warning:

=============================== warnings summary ===============================
tests/test_urls.py::test_404_page_created
tests/test_urls.py::test_404_page_created
  /usr/lib64/python3.8/importlib/__init__.py:127: RemovedInSphinx80Warning: The alias 'sphinx.util.progress_message' is deprecated, use 'sphinx.util.display.progress_message' instead. Check CHANGES for Sphinx API modifications.
    return _bootstrap._gcd_import(name[level:], package, level)

tests/test_urls.py: 27 warnings
  /home/tkloczko/rpmbuild/BUILD/sphinx-notfound-page-0.8.3/notfound/utils.py:34: PendingDeprecationWarning: nodes.Node.traverse() is obsoleted by Node.findall().
    for node in doctree.traverse(nodetype):

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
XFAIL tests/test_urls.py::test_deprecation_warnings - Not sure how to capture warnings from events
FAILED tests/test_urls.py::test_default_settings - assert '<link rel="shortcu...
FAILED tests/test_urls.py::test_default_language_setting - assert '<link rel=...
FAILED tests/test_urls.py::test_default_version_setting - assert '<link rel="...
FAILED tests/test_urls.py::test_urls_prefix_setting - assert '<link rel="shor...
FAILED tests/test_urls.py::test_urls_prefix_setting_none - assert '<link rel=...
FAILED tests/test_urls.py::test_sphinx_resource_urls - assert '<script src="/...
============= 6 failed, 23 passed, 1 xfailed, 29 warnings in 6.52s =============
```
</details>

**Comment by humitos** [comment]
I seems this issue is solved in https://github.com/readthedocs/sphinx-notfound-page/pull/218, right?

**Comment by humitos** [comment]
I released https://pypi.org/project/sphinx-notfound-page/1.0.0rc1/, please give it a try and let me know if it works fine for your use case.

**Comment by humitos** [comment]
I released [v1.0.0](https://pypi.org/project/sphinx-notfound-page/1.0.0/). I hope it works for everybody 🙏🏼
